libcudacxx/include/cuda/std/inplace_vector

```diff
@@ -0,0 +1,1808 @@
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr _CUDA_VRANGES::iterator_t<_Range>
+  try_append_range(_Range&& __range) noexcept(_CCCL_TRAIT(is_nothrow_move_constructible, _Tp))
+  {
+    auto __first = __range.begin();
+    auto __last  = __range.end();
+    for (; this->size() != _Capacity && __first != __last; ++__first)
+    {
+      emplace_back(*__first);
+    }
+    return __first;
+  }
+
+  _LIBCUDACXX_TEMPLATE(class _Range)
+  _LIBCUDACXX_REQUIRES(_CUDA_VRANGES::__container_compatible_range<_Range, _Tp> _LIBCUDACXX_AND
+                         _CUDA_VRANGES::forward_range<_Range> _LIBCUDACXX_AND _CUDA_VRANGES::sized_range<_Range>)
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr _CUDA_VRANGES::iterator_t<_Range>
+  try_append_range(_Range&& __range) noexcept(_CCCL_TRAIT(is_nothrow_move_constructible, _Tp))
+  {
+    _CCCL_IF_CONSTEXPR (_Capacity == 0)
+    {
+      return __range.begin();
+    }
+
+    const auto __capacity = _Capacity - this->size();
+    const auto __size     = _CUDA_VRANGES::size(__range);
+    const auto __diff     = __size < __capacity ? __size : __capacity;
+
+    auto __first  = __range.begin();
+    auto __middle = _CUDA_VRANGES::next(__first, __diff);
+    this->__uninitialized_move(__first, __middle, this->end());
+    return __middle;
+  }
+
+  _LIBCUDACXX_TEMPLATE(class _Range)
+  _LIBCUDACXX_REQUIRES(_CUDA_VRANGES::__container_compatible_range<_Range, _Tp> _LIBCUDACXX_AND
+                         _CUDA_VRANGES::forward_range<_Range> _LIBCUDACXX_AND(!_CUDA_VRANGES::sized_range<_Range>))
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr _CUDA_VRANGES::iterator_t<_Range>
+  try_append_range(_Range&& __range) noexcept(_CCCL_TRAIT(is_nothrow_move_constructible, _Tp))
+  {
+    _CCCL_IF_CONSTEXPR (_Capacity == 0)
+    {
+      return __range.begin();
+    }
+
+    const auto __capacity  = static_cast<ptrdiff_t>(_Capacity - this->size());
+    auto __first           = __range.begin();
+    const auto __size      = static_cast<ptrdiff_t>(_CUDA_VRANGES::distance(__first, __unwrap_end(__range)));
+    const ptrdiff_t __diff = __size < __capacity ? __size : __capacity;
+
+    auto __middle = _CUDA_VRANGES::next(__first, __diff);
+    this->__uninitialized_move(__first, __middle, this->end());
+    return __middle;
+  }
+#  endif // _CCCL_STD_VER >= 2017 && !defined(_CCCL_COMPILER_MSVC_2017)
+
+  using __base::unchecked_emplace_back;
+
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr reference
+  unchecked_push_back(const _Tp& __value) noexcept(_CCCL_TRAIT(is_nothrow_copy_constructible, _Tp))
+  {
+    return this->unchecked_emplace_back(__value);
+  }
+
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr reference
+  unchecked_push_back(_Tp&& __value) noexcept(_CCCL_TRAIT(is_nothrow_move_constructible, _Tp))
+  {
+    return this->unchecked_emplace_back(_CUDA_VSTD::move(__value));
+  }
+
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr void pop_back() noexcept
+  {
+    const auto __end = this->end();
+    this->__destroy(__end, __end + 1);
+  }
```

Comment:
I don't get this. Why isn't this:
```suggestion
  _LIBCUDACXX_INLINE_VISIBILITY constexpr void pop_back() noexcept
  {
    const auto __end = this->end();
    this->__destroy(__end - 1, __end);
  }
```

Reply:
It should be, that is a remnant from a version where __change_size was applied right before